From 92f34d40297a59c470d17c4b64b81b06dfc80097 Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Tue, 22 Dec 2020 21:15:41 +0000
Subject: [PATCH] (internal.c) Fix -Wcast-qual warning

internal.c:400:14: error: assigning to 'char *' from 'const char *' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
    tmp_path = sftpfs_fix_filename (path_element2->path, &tmp_path_len);
             ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
internal.c:601:25: warning: cast from 'const char *' to 'char *' drops const qualifier [-Wcast-qual]
    tmp_path = (char *) sftpfs_fix_filename (path_element2->path, &tmp_path_len);
                        ^

Found by Clang-11

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 src/vfs/sftpfs/internal.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/vfs/sftpfs/internal.c b/src/vfs/sftpfs/internal.c
index baa2de4f7..74a4d5c67 100644
--- a/src/vfs/sftpfs/internal.c
+++ b/src/vfs/sftpfs/internal.c
@@ -391,14 +391,15 @@ sftpfs_symlink (const vfs_path_t * vpath1, const vfs_path_t * vpath2, GError **
     const vfs_path_element_t *path_element1;
     const vfs_path_element_t *path_element2 = NULL;
     char *tmp_path;
+    const char *ctmp_path;
     unsigned int tmp_path_len;
     int res;
 
     if (!sftpfs_op_init (&super, &path_element2, vpath2, mcerror))
         return -1;
 
-    tmp_path = (char *) sftpfs_fix_filename (path_element2->path, &tmp_path_len);
-    tmp_path = g_strndup (tmp_path, tmp_path_len);
+    ctmp_path = sftpfs_fix_filename (path_element2->path, &tmp_path_len);
+    tmp_path = g_strndup (ctmp_path, tmp_path_len);
 
     path_element1 = vfs_path_get_by_index (vpath1, -1);
 
@@ -592,14 +593,15 @@ sftpfs_rename (const vfs_path_t * vpath1, const vfs_path_t * vpath2, GError ** m
     const vfs_path_element_t *path_element1;
     const vfs_path_element_t *path_element2 = NULL;
     char *tmp_path;
+    const char *ctmp_path;
     unsigned int tmp_path_len;
     int res;
 
     if (!sftpfs_op_init (&super, &path_element2, vpath2, mcerror))
         return -1;
 
-    tmp_path = (char *) sftpfs_fix_filename (path_element2->path, &tmp_path_len);
-    tmp_path = g_strndup (tmp_path, tmp_path_len);
+    ctmp_path = sftpfs_fix_filename (path_element2->path, &tmp_path_len);
+    tmp_path = g_strndup (ctmp_path, tmp_path_len);
 
     path_element1 = vfs_path_get_by_index (vpath1, -1);
 
